### ISSUE TITLE
Is there really no such attribute 'exports' in 'haskell_library' rule?

### ISSUE THREAD

**Issue by philderbeast** [tweag/rules_haskell#416]
Following the docs for [haskell_library](https://api.haskell.build/haskell/haskell.html#haskell_library), I tried adding an exports attribute to the tutorial's `booleans` target, setting it initially to the default value, an empty map.

```
haskell_library(
    name = "booleans",
    srcs = ["Bool.hs"],
    visibility = ["//main:__pkg__"],
    exports = {},
)
```

```
tutorial> bazel build //lib:booleans
ERROR: /.../tutorial/lib/BUILD:1:1:
//lib:booleans: no such attribute 'exports' in 'haskell_library' rule
ERROR: error loading package 'lib': Package 'lib' contains errors
INFO: Elapsed time: 0.061s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (1 packages loaded)
```

It certainly looks like the exports attribute should be available;

https://github.com/tweag/rules_haskell/blob/be34f45259ba0d097b62f04fcdad6a8a8b11510a/haskell/haskell.bzl#L187-L188

**Comment by mboes** [comment]
See [the WORKSPACE file](https://github.com/tweag/rules_haskell_examples/blob/master/tutorial/WORKSPACE). It tells Bazel to use a release version of rules_haskell. The `exports` feature is fairly new. So it's probably not part of the latest release.

**Comment by mboes** [comment]
Close?

**Comment by philderbeast** [comment]
I'm trying to see what changes I need to make to be able to use exports in the [vector](https://github.com/tweag/rules_haskell_examples/blob/master/vector/BUILD.bazel) example before I close this issue.

**Comment by mboes** [comment]
That's part of the [Bazel documentation on WORKSPACE files](https://docs.bazel.build/versions/master/build-ref.html#workspace). In this case it would something like:

```diff
diff --git i/tutorial/WORKSPACE w/tutorial/WORKSPACE
index 4ee9385..949a7d9 100644
--- i/tutorial/WORKSPACE
+++ w/tutorial/WORKSPACE
@@ -1,10 +1,12 @@
 workspace(name = "tutorial")
 
+# Whatever is the latest commit.
+RULES_HASKELL_VERSION = "ea52da0835a1ce28a8d7c141dcb7e3ef936fa54b"
+
 http_archive(
     name = "io_tweag_rules_haskell",
-    strip_prefix = "rules_haskell-0.6",
-    urls = ["https://github.com/tweag/rules_haskell/archive/v0.6.tar.gz"],
-    sha256 = "ab0411a7907074e15bc959914eb0642fd44016b89cf1b92e7e6bc9719398e725",
+    strip_prefix = "rules_haskell-{}".format(RULES_HASKELL_VERSION),
+    urls = ["https://github.com/tweag/rules_haskell/archive/{}.tar.gz".format(RULES_HASKELL_VERSION)],
 )
 
 load("@io_tweag_rules_haskell//haskell:repositories.bzl", "haskell_repositories")
```

**Comment by philderbeast** [comment]
I have [applied the fix you suggest](https://github.com/philderbeast/rules_haskell_examples/commit/8cf8a054eb13979f2ad995155984a9c716306ade) using the vector example to try adding `exports = {}`. With this though I'm getting these errors;

```
> bazel build //vector:vector
INFO: SHA256 (https://github.com/tweag/rules_haskell/archive/ea52da0835a1ce28a8d7c141dcb7e3ef936fa54b.tar.gz) = 84038d6f115995121c01c360c806c0356380139ae881425aaef1627bdf7550f8
INFO: Build options have changed, discarding analysis cache.
ERROR: /.../rules_haskell_examples/vector/BUILD.bazel:14:1: every rule of type haskell_library implicitly depends upon the target '@io_tweag_rules_haskell//haskell:assets/ghci_script', but this target could not be found because of: error loading package '@io_tweag_rules_haskell//haskell': Extension file not found. Unable to load package for '@io_bazel_skydoc//skylark:skylark.bzl': The repository could not be resolved
ERROR: /.../rules_haskell_examples/vector/BUILD.bazel:14:1: every rule of type haskell_library implicitly depends upon the target '@io_tweag_rules_haskell//haskell:private/ghci_repl_wrapper.sh', but this target could not be found because of: error loading package '@io_tweag_rules_haskell//haskell': Extension file not found. Unable to load package for '@io_bazel_skydoc//skylark:skylark.bzl': The repository could not be resolved
ERROR: /.../rules_haskell_examples/vector/BUILD.bazel:14:1: every rule of type haskell_library implicitly depends upon the target '@io_tweag_rules_haskell//haskell:ls_modules', but this target could not be found because of: error loading package '@io_tweag_rules_haskell//haskell': Extension file not found. Unable to load package for '@io_bazel_skydoc//skylark:skylark.bzl': The repository could not be resolved
ERROR: Analysis of target '//vector:vector' failed; build aborted: error loading package '@io_tweag_rules_haskell//haskell': Extension file not found. Unable to load package for '@io_bazel_skydoc//skylark:skylark.bzl': The repository could not be resolved
INFO: Elapsed time: 0.861s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (6 packages loaded)
    currently loading: @io_tweag_rules_haskell//haskell
```

**Comment by lunaris** [comment]
I had this issue a few days ago and worked around it by using a local copy of rules_haskell in which I'd removed the Skydoc imports and target that were added in the documentation patches added a week or so ago (sorry on mobile so not got an exact reference handy). Not sure why this issue isn't affecting all users because I couldn't find a way to sort it. Unless it's Hazel related?

**Comment by mboes** [comment]
Ah yes, that's a regression introduced in 86d15e1ff72f46e89da3476a53f7cce9c728dbb7, which I spotted this morning. Use 6b2978a1d3403c9494ba37061d3985e90c7df90a for the time being.

**Comment by mboes** [comment]
@lunaris this issue only shows up for *users* of rules_haskell apparently, but not in rules_haskell itself (which is why CI didn't catch it). That is, it only appears in workspaces that import rules_haskell. And even then, only if the skydoc repo was not also imported.

**Comment by philderbeast** [comment]
I got the build working with `exports = {}` by also [importing the skydoc repo](https://github.com/philderbeast/rules_haskell_examples/commit/5651321cc9d1faec270e4979551f2ac3b70003a2).

**Comment by philderbeast** [comment]
I can confirm too that with the fix for #419, I can [remove the import of skydocs](https://github.com/philderbeast/rules_haskell_examples/commit/d704a5a15809955762022b14bc0a72664e812a4f) and the build works.
